core/src/main/java/org/elasticsearch/http/netty/NettyHttpServerTransport.java

```diff
@@ -274,7 +269,7 @@ public NettyHttpServerTransport(Settings settings, NetworkService networkService
             maxContentLength = new ByteSizeValue(100, ByteSizeUnit.MB);
         }
         this.maxContentLength = maxContentLength;
-        this.inFlightRequestsBreaker = circuitBreakerService.getBreaker(CircuitBreaker.IN_FLIGHT_REQUESTS);
+        this.circuitBreakerService = circuitBreakerService;
```

Comment:
why not move all of this to the HttpServer? then we don't need to change the interface and keep things local..